Update ANGLE to 2024-07-31 (3e83d1887e2bad0d5dac01b37fa94b88188ecd4f)

kkinnunen-apple opened the pull request:
#### 93cee279db40919361e72b314613badcc15d1ad7
<pre>
Update ANGLE to 2024-07-31 (3e83d1887e2bad0d5dac01b37fa94b88188ecd4f)
<a href="https://bugs.webkit.org/show_bug.cgi?id=277410">https://bugs.webkit.org/show_bug.cgi?id=277410</a>
<a href="https://rdar.apple.com/problem/132885219">rdar://problem/132885219</a>

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline 0c7331dfe71de0a3afc85106dce8b4948adbd5dd..3e83d1887e2bad0d5dac01b37fa94b88188ecd4f --pretty=%h %s
3e83d1887e Roll Chromium from 230f7abd1f6d to cedeeaab3123 (601 revisions)
65f08d0851 Vulkan: Work around driver bug with nested switch
346ba328ea Vulkan: Simplify pulling samplers out of structs
451d78d996 Prune trivial infinite loops in WebGL contexts
4c883359f5 Add dma_buf build support to generate_android_bp
3cda352736 Capture/Replay: fix CaptureProducedRequiredFiles bottleneck
e413ecafb1 Increase parallelism Win Intel UHD 770
198de0447a Metal: fix wrong render encoder&apos;s serial assignment
2bc1519ccc Metal: fix base/max level change for mutable textures.
aaf39d92ba FrameCapture: fix (nil) showing up in linux-trace capture
7e70034400 Roll SwiftShader from c4dfa69de7de to 01b188e56477 (1 revision)
15b354d50e Unsuppress ASAN/UBSAN suppressions for fixed tests
3b5e09ae54 Roll Chromium from 232a6a5422d1 to 230f7abd1f6d (544 revisions)
0ef0668746 Stage the angle_commit.h for commit during ANGLE Android Roll
da86429482 Capture/Replay: rework capture_replay_tests, enable linux-trace
bd3a3308f0 Vulkan: Remove implicit image barrier for shader write
c974461757 Fix Unknown Android Git Hash in Android Builds
554cc4845f Tests: end2end tests process test args before TestSuite()
568caaa048 Prune switch(constant) with no matching case
b8cc14d709 skip lib starting with &quot;//&quot; when generating Android bp
aa8e1e5030 Roll VK-GL-CTS from c66570e674d0 to 3d152263d502 (6 revisions)
506cb7c18d Roll Chromium from 2e8f18571b3e to 232a6a5422d1 (733 revisions)
572025841d Vulkan: Fix dispatch-after-closed-render-pass bug
f47a882a15 Cleanup: remove vulkan_clean_old_validation_layer_objects
b20cd34ddd Add build flag for a &quot;dma_buf&quot; Android build
a0a832de0a Revert &quot;GL: Forward client-side arrays to the driver when possible&quot;
31b2bcd6e3 Roll vulkan-deps from 49b5e420b19a to 068847956e41 (2 revisions)
2c1d78ef80 Roll Chromium from 15e30105c06e to 2e8f18571b3e (242 revisions)
d306e0488f Tests: Add Shovel Knight Pocket Dungeon trace
ca6238fb00 Manual roll Chromium from 99513aa4b41c to 15e30105c06e (813 revisions)
a6c2b43465 GL: Forward client-side arrays to the driver when possible
f440d17cb2 Manual roll Chromium from 834617e372fd to 99513aa4b41c (90 revisions)
48d19755a3 Manual roll VK-GL-CTS from 5a542d2049d7 to c66570e674d0 (8 revisions)
f184de3fc6 Don&apos;t roll third_party/rust-toolchain into AOSP
81086a18b8 Manual roll VK-GL-CTS from 5b2dfe7c775a to 5a542d2049d7 (1 revision)
5013e1998f Roll vulkan-deps from 8e90204125ac to 49b5e420b19a (8 revisions)
4f498eaa14 WebGPU: Add more format support
84e54d885f WebGPU: Add command buffers and renderpass management
703c59601c Vulkan: PPO must warmup pipeline cache according to GPL support
d10d1c4f12 Manual roll Chromium from 04132ee8f8a5 to 834617e372fd (557 revisions)
6afff89309 Roll Rust tools with Chromium
5a44aa26a0 Roll vulkan-deps from 5d0bdd3702f7 to 8e90204125ac (4 revisions)
fc381db465 Metal: Treat link jobs as thread-safe
b98c4d8107 FrameCapture: Add restore of VAO buffer binding
6eda255762 Metal: Pass an existing context to CompileMslTask
85264808d0 tls.h: check ANGLE_USE_ANDROID_TLS_SLOT to define GL TLS
1b0e74f377 Manual roll Chromium from 9739f7f11240 to 04132ee8f8a5 (755 revisions)
7b0dc666b4 Vulkan: Enable explicitlyEnablePerSampleShading for most vendors
2f05b993d1 Roll vulkan-deps from be96f8cab214 to 5d0bdd3702f7 (7 revisions)
7691cea7e5 Vulkan: Remove seamful cubemap emulation
0d458614fb Vulkan: Fix PBO readbacks with small row length
975674ea9d Metal: Fix PBO readback failures with small row length
e37c79d437 Fix DecompressBlob() when compressed data is invalid
fb4cdb18bb Roll vulkan-deps from edbbcd521ef6 to be96f8cab214 (9 revisions)
d44a303297 Roll Chromium from 06c75e7c6658 to 9739f7f11240 (859 revisions)
85edb90719 Vulkan: Remove the adjustClearColorPrecision feature for new driver
93491ae30b FrameCapture: Handle active texture in MEC
1fe50b446b Make implicit `this` capture explicit
327796df8c FrameCapture: Add a new CapturedTest for glActiveTexture
0364ac4ad8 Set up ANGLE shadow buckets
fa940e3628 Roll vulkan-deps from 4b8a413b6b64 to edbbcd521ef6 (4 revisions)
16376d10ea Roll Chromium from b5eaffbef1cb to 06c75e7c6658 (597 revisions)
0e598a0c3f FrameCapture: Unbind PIXEL_UNPACK_BUFFER for texture restore
98ff1c14b2 FrameCapture: Improve buffer binding reset
a5db20f749 Traces: Switch to regex for power rails
80149a7f6e Replace variable-length array with vector
97a249b71f Manual roll Chromium from 71adbebaf5b6 to b5eaffbef1cb (807 revisions)
86b22745bb Vulkan: Fix mPipelineCache is not initialized after terminate
6c585f8570 Roll vulkan-deps from ace59194583a to 4b8a413b6b64 (10 revisions)
62d9f847e4 Speed up UniformBuffer support on Metal
9968e98cd4 Metal: Fix compute path for 16-bit PBO readbacks
43dc24dd15 Skip couple non-deterministic angle_restricted_trace_gold_tests
1db80b8808 Reland &quot;Vulkan: Use VK_KHR_dynamic_rendering[_local_read]&quot;
3fe6cae17a Fix gl::roundToNearest on x86
3c21ff7d35 Roll vulkan-deps from 99f840c7c7ad to ace59194583a (12 revisions)
f45494ee39 Roll SwiftShader from 389854967d78 to c4dfa69de7de (1 revision)
2452e62d43 Roll Chromium from 4be473888de4 to 71adbebaf5b6 (678 revisions)
1f87cbc925 Vulkan: Fix late-added resolve attachment tracking
468fcbe033 Manual roll VK-GL-CTS from 43432bf265e3 to 5b2dfe7c775a (17 revisions)
3540f8e892 Fix gl::floatToNormalized for negative inputs
f9af6f1cc3 Add ReadPixelsTextureNorm16PBOTest
c4274d549c WebGPU: Add pipeline creation and caching.
58a56ca473 Vulkan: Restrict VkEvent to TBRs
686ed56d3d End ANGLE Mac Intel 14.5 experiment
84c77437db GL: Disable GL_KHR_blend_equation_advanced on various devices
9880af58e3 GL: Fix map input buffer out of range in streamAttributes
53ada6b4af Roll vulkan-deps from 7dc21774dc62 to 99f840c7c7ad (13 revisions)
b8d21cd230 Roll SwiftShader from f23c77132e0b to 389854967d78 (1 revision)
9e60edc7ca Roll Chromium from dcc3741ff616 to 4be473888de4 (519 revisions)
85875b9676 Fix validation regarding cube map array dims
a8eded7713 Vulkan: remove unused function
faae3c32c0 Add necessary MSAN include when ANGLE_WITH_MSAN
43ecf2bd98 dEQP tests: skip comp_swap r32i/r32ui on Pixel 6
06921c61aa vulkan: filter out devices with insufficient API version
56861ff18d Manual roll Chromium from d3a8a2da6ae3 to dcc3741ff616 (894 revisions)
e6d43dc19b Roll vulkan-deps from 9563c7bbc6c9 to 7dc21774dc62 (15 revisions)
df90bbc50d Refactoring: move angle::HashMap and HashSet to own header
2997e3545e Start Win Intel UHD 770 experiment
a5161f01b4 Vulkan: Enable event based image barrier
24322b7d29 Vulkan: feature for cached non-coherent for dyn/stream buffers
8ed4d42734 Metal: Remove link tasks dependency on context
245c12436f Revert &quot;Disallow read color type conversions for norm16 formats&quot;
d89a205d35 Roll vulkan-deps from bc656cdad31e to 9563c7bbc6c9 (11 revisions)
298d55f321 Roll Chromium from 51ac27c135bf to d3a8a2da6ae3 (571 revisions)
c5ab1cebcc Metal: Remove library creation dependency on ContextDevice
a93476a1ce D3D11: Pass 3D slice index to Image11::copyWithoutConversion
6578b9c04b Vulkan: Exclude compute/preFrag only access images from event
402c8ccd88 Vulkan: Limit VkEvent for images that has fragment access only
eda67d7ee8 Avoid including fstream from angleutils.h
a15d4f977c Roll Chromium from 5748d73603a8 to 51ac27c135bf (686 revisions)
e59ae389f7 angle_enable_wgpu requires X11 on Linux
126275cbff Roll vulkan-deps from 67847f196c19 to bc656cdad31e (7 revisions)
87e9ee9c2b egl: add missing extension string for D3D LUID
584fbcee1e Vulkan: Rework swap-time barrier logic
373ac5411a Vulkan: Make surface RP check independent from framebuffer object
f7620039be Vulkan: Prepare syncval suppressions for dynamic rendering
590010b690 D3D11: Cleanup stale CopyTexImageTestES3 suppressions
867697b7ad Vulkan: Add ImageHelper::onRenderPassAttach helper function
9ca3ed375c Vulkan: Let ContextVk::onResourceAccess uses retainImage
95e6630754 Resolve compilation during program link resolve
c62c98dfaa Revert &quot;Temporarily angle_enable_vulkan_api_dump_layer default false&quot;
7d461b21a4 Revert &quot;Vulkan: Use VK_KHR_dynamic_rendering[_local_read]&quot;
e208994a09 Add a note about std::aligned_alloc caveats
b0bb41865e Roll vulkan-deps from 1d4b1061b004 to 67847f196c19 (3 revisions)
e5b442dedd Disallow read color type conversions for norm16 formats
13c93ca96f Roll Chromium from f0fb088cc7f4 to 5748d73603a8 (622 revisions)
0679b3eb2f Manual roll vulkan-deps from 17d345a0f23d to 1d4b1061b004 (24 revisions)
6052d0f4ac Vulkan: Fix EGL_EXT_buffer_age with single buffered surface
21fc5e8773 Move loadimage SSE includes/defines out of common headers
c68c7dca38 Roll Chromium from b1148c13c7a9 to f0fb088cc7f4 (639 revisions)
c379ff4804 Vulkan: Use VK_KHR_dynamic_rendering[_local_read]
2763d52052 Do not rely on 16-bit color read conversions in tests, Part 2
8c546d356c Vulkan: Limit VkEvent for usage matters for Manhattan31 only
a68351cb9e vulkan: add workround for speical shadowmap at intel device.
bc38b7ade7 Start ANGLE Intel macOS 14.5 experiment
0a7e1daf62 Tests: fix MultithreadFenceDraw, MultithreadFenceTexImage
640e371612 Do not rely on 16-bit color read conversions in tests
3c8a8f7ad9 Temporarily angle_enable_vulkan_api_dump_layer default false
2a87db6987 Vulkan: Remove unused render pass closure reason
6c6975afe5 Manual roll vulkan-deps from 5fdd3da87369 to 17d345a0f23d (10 revisions)
66f4b76c4e Roll Chromium from a98f22a2505a to b1148c13c7a9 (354 revisions)
db00dee577 Vulkan: Fix SPIR-V gen for literally indexed swizzles
58dc069c02 Rename 16-bit EXPECT_PIXEL test helpers
1dd3a4c405 D3D11: Use built-in position for gl_FragCoord.z on SM 4.0+
f22189ecfe Roll chromium_revision c174259bf6..a98f22a250 (1323232:1323712)
7489c5f0b9 Roll SwiftShader from a0ec371d8331 to f23c77132e0b (1 revision)
c46d94d18b Updating ceval&apos;s README.chromium to include Revision
fe02de2504 Metal: Remove MTLGetMSL dependency on context
dfec7f4c5b Updating clspv&apos;s README.chromium to include Revision
cb0b851f14 Add Revision: field to third party
fcf3a1c05f GL: Allow shader compilation with cached translated source
7d61980e65 Vulkan: Rename DescriptorSetLayoutDesc update() to addBinding()
a4f3f6bfa1 GL: Simplify members of ShaderGL.
b7d4c5d1d4 Metal: Disable fast math if isinf or isnan are used
397eb7c81e Vulkan: Easier to read SPIR-V disassembly
d66ac3584a Roll Chromium from 34f452e68b67 to c174259bf610 (620 revisions)
f2ee319dda Roll Chromium from 51d79108bbb3 to 34f452e68b67 (970 revisions)
b954755f4e Release all memory allocated in TCompiler::compile
b645633438 Reland: GL: Support KHR_blend_equation_advanced
d3987e50be Roll vulkan-deps from 5faad80a00af to 5fdd3da87369 (6 revisions)
2efce9babc Roll chromium_revision 40020cda88..51d79108bb (1320198:1321642)
ef2c966502 GL: Fix dispatch table for desktop-only entry points
78ee33e4ef Roll third_party/dawn/ 6cdf3a1a1..eb249b37e (310 commits; 3 trivial rolls)
d6bd2c7028 Roll vulkan-deps from 4a2ff9e87700 to 5faad80a00af (6 revisions)
d8700dc951 Avoid ANGLE vulkantools dependency in Dawn build
8667fb3e5a Roll vulkan-deps from 2796aaff7eef to 4a2ff9e87700 (10 revisions)
2a2b5cbd70 Roll VK-GL-CTS from 25d8c0099575 to 43432bf265e3 (8 revisions)
4f23429a1f egl_angle_ext: fix incorrect EGL_ANGLE_device_d3d11 name
1062271d99 Roll vulkan-deps from 7ff358e64e2b to 2796aaff7eef (10 revisions)
e86ba9abe9 Vulkan: Fix querying support of prim restart patch
3813e805b1 Add env var/property control of API dump layer
a674dc1dae Unify updating cached color/read render targets
d939444738 Roll vulkan-deps from d2e24d980359 to 7ff358e64e2b (2 revisions)
48ebe74024 Roll Chromium from 44cf5c690e8b to 40020cda886f (613 revisions)
46dd6457f4 Vulkan: Use DONT_CARE ops for missing D/S aspects
35f01e7f93 ESSL -&gt; WGSL: Support basic control flow
89ef2812d4 ESSL -&gt; WGSL: unary and binary exprs, operators, array access
4f8c42de40 Manual roll VK-GL-CTS from c201252e6fbe to 25d8c0099575 (13 revisions)
3e99741974 Add format conversions for WebGPU formats.
0bf157f12b Tests: Update expectations for new deqp test name
85e2e6a912 WebGPU: Add Window and Pbuffer surfaces.
cd9f457ffb Roll vulkan-deps from 2e4ed1cfe73c to d2e24d980359 (13 revisions)
7d45aac98f Roll SwiftShader from f65e518550a8 to a0ec371d8331 (1 revision)
bdce308b96 Roll Chromium from 9df67ea3cc71 to 44cf5c690e8b (618 revisions)
0463273435 Add EGL query option for minor version
df7bda5c11 Disable MSRTSS if not supported for RGBA
8b1daa18e9 Vulkan: Fix MSRTT failure from undef addnl format
0c5b327a32 Extend advanced blend tests to cover core and KHR
0ccc018e8f Reland GL: Avoid calling glBindFragDatalocationIndexed on Qualcomm
7ca9f46a77 ESSL -&gt; WGSL: Emit func calls, struct access, constants
016ef60d5f ValidateAST asserts when refers to builtIn struct type name
228f56b853 Roll SwiftShader from de870ac7518f to f65e518550a8 (1 revision)
84f88f0032 Roll vulkan-deps from bf2b4e8ba30e to 2e4ed1cfe73c (4 revisions)
d9af242315 Roll Chromium from dd0b2a0bee86 to 9df67ea3cc71 (513 revisions)
dd25e193f2 Add a texture copy test that does multiple layer copies
beb75984ca Fix incorrect verification for SimultaneousUploadAndDraw
e095c06f09 Suppress end2end MultisampledRenderToTexture* failures
6bfd7d139b Manual roll vulkan-deps from c0c0899c3614 to bf2b4e8ba30e (5 revisions)
15b95b4a5d Skip WebGL2UniformBufferTest.LargeArrayOfStructs
5e64ccf9c4 Roll SwiftShader from cea33ab2d5ad to de870ac7518f (1 revision)
45b67e0584 Roll vulkan-deps from 30f8ac1a3f18 to c0c0899c3614 (6 revisions)

Canonical link: <a href="https://commits.webkit.org/281691@main">https://commits.webkit.org/281691@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/59aeea7db81a6ee393fb8ad8910645dcd6006464

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60721 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40080 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13297 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64652 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/11268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47756 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11543 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49097 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/11268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62755 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/37327 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/52600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/29928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/34015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/9832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10181 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/55857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/10130 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66381 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4665 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9963 "Build is in progress. Recent messages:") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4686 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/52574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/56646 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/88/builds/3864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9125 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/35885 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/36967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/38060 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36712 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/ThirdParty/ANGLE/.gitignore:
```diff
@@ -123,6 +124,7 @@
 /tools/perf
 /tools/protoc_wrapper
 /tools/python
+/tools/rust
```
!!

djg commented on Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 55;
+	objectVersion = 54;
```
Do we know what causes this unnecessary churn on this property?

kkinnunen-apple replied:
I think it's different xcode versions, but I thought mine was up-to-date.

djg reviewed (approved):
:shipit:

kkinnunen-apple committed:
Update ANGLE to 2024-07-31 (3e83d1887e2bad0d5dac01b37fa94b88188ecd4f)
https://bugs.webkit.org/show_bug.cgi?id=277410
rdar://problem/132885219

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline 0c7331dfe71de0a3afc85106dce8b4948adbd5dd..3e83d1887e2bad0d5dac01b37fa94b88188ecd4f --pretty=%h %s
3e83d1887e Roll Chromium from 230f7abd1f6d to cedeeaab3123 (601 revisions)
65f08d0851 Vulkan: Work around driver bug with nested switch
346ba328ea Vulkan: Simplify pulling samplers out of structs
451d78d996 Prune trivial infinite loops in WebGL contexts
4c883359f5 Add dma_buf build support to generate_android_bp
3cda352736 Capture/Replay: fix CaptureProducedRequiredFiles bottleneck
e413ecafb1 Increase parallelism Win Intel UHD 770
198de0447a Metal: fix wrong render encoder's serial assignment
2bc1519ccc Metal: fix base/max level change for mutable textures.
aaf39d92ba FrameCapture: fix (nil) showing up in linux-trace capture
7e70034400 Roll SwiftShader from c4dfa69de7de to 01b188e56477 (1 revision)
15b354d50e Unsuppress ASAN/UBSAN suppressions for fixed tests
3b5e09ae54 Roll Chromium from 232a6a5422d1 to 230f7abd1f6d (544 revisions)
0ef0668746 Stage the angle_commit.h for commit during ANGLE Android Roll
da86429482 Capture/Replay: rework capture_replay_tests, enable linux-trace
bd3a3308f0 Vulkan: Remove implicit image barrier for shader write
c974461757 Fix Unknown Android Git Hash in Android Builds
554cc4845f Tests: end2end tests process test args before TestSuite()
568caaa048 Prune switch(constant) with no matching case
b8cc14d709 skip lib starting with "//" when generating Android bp
aa8e1e5030 Roll VK-GL-CTS from c66570e674d0 to 3d152263d502 (6 revisions)
506cb7c18d Roll Chromium from 2e8f18571b3e to 232a6a5422d1 (733 revisions)
572025841d Vulkan: Fix dispatch-after-closed-render-pass bug
f47a882a15 Cleanup: remove vulkan_clean_old_validation_layer_objects
b20cd34ddd Add build flag for a "dma_buf" Android build
a0a832de0a Revert "GL: Forward client-side arrays to the driver when possible"
31b2bcd6e3 Roll vulkan-deps from 49b5e420b19a to 068847956e41 (2 revisions)
2c1d78ef80 Roll Chromium from 15e30105c06e to 2e8f18571b3e (242 revisions)
d306e0488f Tests: Add Shovel Knight Pocket Dungeon trace
ca6238fb00 Manual roll Chromium from 99513aa4b41c to 15e30105c06e (813 revisions)
a6c2b43465 GL: Forward client-side arrays to the driver when possible
f440d17cb2 Manual roll Chromium from 834617e372fd to 99513aa4b41c (90 revisions)
48d19755a3 Manual roll VK-GL-CTS from 5a542d2049d7 to c66570e674d0 (8 revisions)
f184de3fc6 Don't roll third_party/rust-toolchain into AOSP
81086a18b8 Manual roll VK-GL-CTS from 5b2dfe7c775a to 5a542d2049d7 (1 revision)
5013e1998f Roll vulkan-deps from 8e90204125ac to 49b5e420b19a (8 revisions)
4f498eaa14 WebGPU: Add more format support
84e54d885f WebGPU: Add command buffers and renderpass management
703c59601c Vulkan: PPO must warmup pipeline cache according to GPL support
d10d1c4f12 Manual roll Chromium from 04132ee8f8a5 to 834617e372fd (557 revisions)
6afff89309 Roll Rust tools with Chromium
5a44aa26a0 Roll vulkan-deps from 5d0bdd3702f7 to 8e90204125ac (4 revisions)
fc381db465 Metal: Treat link jobs as thread-safe
b98c4d8107 FrameCapture: Add restore of VAO buffer binding
6eda255762 Metal: Pass an existing context to CompileMslTask
85264808d0 tls.h: check ANGLE_USE_ANDROID_TLS_SLOT to define GL TLS
1b0e74f377 Manual roll Chromium from 9739f7f11240 to 04132ee8f8a5 (755 revisions)
7b0dc666b4 Vulkan: Enable explicitlyEnablePerSampleShading for most vendors
2f05b993d1 Roll vulkan-deps from be96f8cab214 to 5d0bdd3702f7 (7 revisions)
7691cea7e5 Vulkan: Remove seamful cubemap emulation
0d458614fb Vulkan: Fix PBO readbacks with small row length
975674ea9d Metal: Fix PBO readback failures with small row length
e37c79d437 Fix DecompressBlob() when compressed data is invalid
fb4cdb18bb Roll vulkan-deps from edbbcd521ef6 to be96f8cab214 (9 revisions)
d44a303297 Roll Chromium from 06c75e7c6658 to 9739f7f11240 (859 revisions)
85edb90719 Vulkan: Remove the adjustClearColorPrecision feature for new driver
93491ae30b FrameCapture: Handle active texture in MEC
1fe50b446b Make implicit `this` capture explicit
327796df8c FrameCapture: Add a new CapturedTest for glActiveTexture
0364ac4ad8 Set up ANGLE shadow buckets
fa940e3628 Roll vulkan-deps from 4b8a413b6b64 to edbbcd521ef6 (4 revisions)
16376d10ea Roll Chromium from b5eaffbef1cb to 06c75e7c6658 (597 revisions)
0e598a0c3f FrameCapture: Unbind PIXEL_UNPACK_BUFFER for texture restore
98ff1c14b2 FrameCapture: Improve buffer binding reset
a5db20f749 Traces: Switch to regex for power rails
80149a7f6e Replace variable-length array with vector
97a249b71f Manual roll Chromium from 71adbebaf5b6 to b5eaffbef1cb (807 revisions)
86b22745bb Vulkan: Fix mPipelineCache is not initialized after terminate
6c585f8570 Roll vulkan-deps from ace59194583a to 4b8a413b6b64 (10 revisions)
62d9f847e4 Speed up UniformBuffer support on Metal
9968e98cd4 Metal: Fix compute path for 16-bit PBO readbacks
43dc24dd15 Skip couple non-deterministic angle_restricted_trace_gold_tests
1db80b8808 Reland "Vulkan: Use VK_KHR_dynamic_rendering[_local_read]"
3fe6cae17a Fix gl::roundToNearest on x86
3c21ff7d35 Roll vulkan-deps from 99f840c7c7ad to ace59194583a (12 revisions)
f45494ee39 Roll SwiftShader from 389854967d78 to c4dfa69de7de (1 revision)
2452e62d43 Roll Chromium from 4be473888de4 to 71adbebaf5b6 (678 revisions)
1f87cbc925 Vulkan: Fix late-added resolve attachment tracking
468fcbe033 Manual roll VK-GL-CTS from 43432bf265e3 to 5b2dfe7c775a (17 revisions)
3540f8e892 Fix gl::floatToNormalized for negative inputs
f9af6f1cc3 Add ReadPixelsTextureNorm16PBOTest
c4274d549c WebGPU: Add pipeline creation and caching.
58a56ca473 Vulkan: Restrict VkEvent to TBRs
686ed56d3d End ANGLE Mac Intel 14.5 experiment
84c77437db GL: Disable GL_KHR_blend_equation_advanced on various devices
9880af58e3 GL: Fix map input buffer out of range in streamAttributes
53ada6b4af Roll vulkan-deps from 7dc21774dc62 to 99f840c7c7ad (13 revisions)
b8d21cd230 Roll SwiftShader from f23c77132e0b to 389854967d78 (1 revision)
9e60edc7ca Roll Chromium from dcc3741ff616 to 4be473888de4 (519 revisions)
85875b9676 Fix validation regarding cube map array dims
a8eded7713 Vulkan: remove unused function
faae3c32c0 Add necessary MSAN include when ANGLE_WITH_MSAN
43ecf2bd98 dEQP tests: skip comp_swap r32i/r32ui on Pixel 6
06921c61aa vulkan: filter out devices with insufficient API version
56861ff18d Manual roll Chromium from d3a8a2da6ae3 to dcc3741ff616 (894 revisions)
e6d43dc19b Roll vulkan-deps from 9563c7bbc6c9 to 7dc21774dc62 (15 revisions)
df90bbc50d Refactoring: move angle::HashMap and HashSet to own header
2997e3545e Start Win Intel UHD 770 experiment
a5161f01b4 Vulkan: Enable event based image barrier
24322b7d29 Vulkan: feature for cached non-coherent for dyn/stream buffers
8ed4d42734 Metal: Remove link tasks dependency on context
245c12436f Revert "Disallow read color type conversions for norm16 formats"
d89a205d35 Roll vulkan-deps from bc656cdad31e to 9563c7bbc6c9 (11 revisions)
298d55f321 Roll Chromium from 51ac27c135bf to d3a8a2da6ae3 (571 revisions)
c5ab1cebcc Metal: Remove library creation dependency on ContextDevice
a93476a1ce D3D11: Pass 3D slice index to Image11::copyWithoutConversion
6578b9c04b Vulkan: Exclude compute/preFrag only access images from event
402c8ccd88 Vulkan: Limit VkEvent for images that has fragment access only
eda67d7ee8 Avoid including fstream from angleutils.h
a15d4f977c Roll Chromium from 5748d73603a8 to 51ac27c135bf (686 revisions)
e59ae389f7 angle_enable_wgpu requires X11 on Linux
126275cbff Roll vulkan-deps from 67847f196c19 to bc656cdad31e (7 revisions)
87e9ee9c2b egl: add missing extension string for D3D LUID
584fbcee1e Vulkan: Rework swap-time barrier logic
373ac5411a Vulkan: Make surface RP check independent from framebuffer object
f7620039be Vulkan: Prepare syncval suppressions for dynamic rendering
590010b690 D3D11: Cleanup stale CopyTexImageTestES3 suppressions
867697b7ad Vulkan: Add ImageHelper::onRenderPassAttach helper function
9ca3ed375c Vulkan: Let ContextVk::onResourceAccess uses retainImage
95e6630754 Resolve compilation during program link resolve
c62c98dfaa Revert "Temporarily angle_enable_vulkan_api_dump_layer default false"
7d461b21a4 Revert "Vulkan: Use VK_KHR_dynamic_rendering[_local_read]"
e208994a09 Add a note about std::aligned_alloc caveats
b0bb41865e Roll vulkan-deps from 1d4b1061b004 to 67847f196c19 (3 revisions)
e5b442dedd Disallow read color type conversions for norm16 formats
13c93ca96f Roll Chromium from f0fb088cc7f4 to 5748d73603a8 (622 revisions)
0679b3eb2f Manual roll vulkan-deps from 17d345a0f23d to 1d4b1061b004 (24 revisions)
6052d0f4ac Vulkan: Fix EGL_EXT_buffer_age with single buffered surface
21fc5e8773 Move loadimage SSE includes/defines out of common headers
c68c7dca38 Roll Chromium from b1148c13c7a9 to f0fb088cc7f4 (639 revisions)
c379ff4804 Vulkan: Use VK_KHR_dynamic_rendering[_local_read]
2763d52052 Do not rely on 16-bit color read conversions in tests, Part 2
8c546d356c Vulkan: Limit VkEvent for usage matters for Manhattan31 only
a68351cb9e vulkan: add workround for speical shadowmap at intel device.
bc38b7ade7 Start ANGLE Intel macOS 14.5 experiment
0a7e1daf62 Tests: fix MultithreadFenceDraw, MultithreadFenceTexImage
640e371612 Do not rely on 16-bit color read conversions in tests
3c8a8f7ad9 Temporarily angle_enable_vulkan_api_dump_layer default false
2a87db6987 Vulkan: Remove unused render pass closure reason
6c6975afe5 Manual roll vulkan-deps from 5fdd3da87369 to 17d345a0f23d (10 revisions)
66f4b76c4e Roll Chromium from a98f22a2505a to b1148c13c7a9 (354 revisions)
db00dee577 Vulkan: Fix SPIR-V gen for literally indexed swizzles
58dc069c02 Rename 16-bit EXPECT_PIXEL test helpers
1dd3a4c405 D3D11: Use built-in position for gl_FragCoord.z on SM 4.0+
f22189ecfe Roll chromium_revision c174259bf6..a98f22a250 (1323232:1323712)
7489c5f0b9 Roll SwiftShader from a0ec371d8331 to f23c77132e0b (1 revision)
c46d94d18b Updating ceval's README.chromium to include Revision
fe02de2504 Metal: Remove MTLGetMSL dependency on context
dfec7f4c5b Updating clspv's README.chromium to include Revision
cb0b851f14 Add Revision: field to third party
fcf3a1c05f GL: Allow shader compilation with cached translated source
7d61980e65 Vulkan: Rename DescriptorSetLayoutDesc update() to addBinding()
a4f3f6bfa1 GL: Simplify members of ShaderGL.
b7d4c5d1d4 Metal: Disable fast math if isinf or isnan are used
397eb7c81e Vulkan: Easier to read SPIR-V disassembly
d66ac3584a Roll Chromium from 34f452e68b67 to c174259bf610 (620 revisions)
f2ee319dda Roll Chromium from 51d79108bbb3 to 34f452e68b67 (970 revisions)
b954755f4e Release all memory allocated in TCompiler::compile
b645633438 Reland: GL: Support KHR_blend_equation_advanced
d3987e50be Roll vulkan-deps from 5faad80a00af to 5fdd3da87369 (6 revisions)
2efce9babc Roll chromium_revision 40020cda88..51d79108bb (1320198:1321642)
ef2c966502 GL: Fix dispatch table for desktop-only entry points
78ee33e4ef Roll third_party/dawn/ 6cdf3a1a1..eb249b37e (310 commits; 3 trivial rolls)
d6bd2c7028 Roll vulkan-deps from 4a2ff9e87700 to 5faad80a00af (6 revisions)
d8700dc951 Avoid ANGLE vulkantools dependency in Dawn build
8667fb3e5a Roll vulkan-deps from 2796aaff7eef to 4a2ff9e87700 (10 revisions)
2a2b5cbd70 Roll VK-GL-CTS from 25d8c0099575 to 43432bf265e3 (8 revisions)
4f23429a1f egl_angle_ext: fix incorrect EGL_ANGLE_device_d3d11 name
1062271d99 Roll vulkan-deps from 7ff358e64e2b to 2796aaff7eef (10 revisions)
e86ba9abe9 Vulkan: Fix querying support of prim restart patch
3813e805b1 Add env var/property control of API dump layer
a674dc1dae Unify updating cached color/read render targets
d939444738 Roll vulkan-deps from d2e24d980359 to 7ff358e64e2b (2 revisions)
48ebe74024 Roll Chromium from 44cf5c690e8b to 40020cda886f (613 revisions)
46dd6457f4 Vulkan: Use DONT_CARE ops for missing D/S aspects
35f01e7f93 ESSL -> WGSL: Support basic control flow
89ef2812d4 ESSL -> WGSL: unary and binary exprs, operators, array access
4f8c42de40 Manual roll VK-GL-CTS from c201252e6fbe to 25d8c0099575 (13 revisions)
3e99741974 Add format conversions for WebGPU formats.
0bf157f12b Tests: Update expectations for new deqp test name
85e2e6a912 WebGPU: Add Window and Pbuffer surfaces.
cd9f457ffb Roll vulkan-deps from 2e4ed1cfe73c to d2e24d980359 (13 revisions)
7d45aac98f Roll SwiftShader from f65e518550a8 to a0ec371d8331 (1 revision)
bdce308b96 Roll Chromium from 9df67ea3cc71 to 44cf5c690e8b (618 revisions)
0463273435 Add EGL query option for minor version
df7bda5c11 Disable MSRTSS if not supported for RGBA
8b1daa18e9 Vulkan: Fix MSRTT failure from undef addnl format
0c5b327a32 Extend advanced blend tests to cover core and KHR
0ccc018e8f Reland GL: Avoid calling glBindFragDatalocationIndexed on Qualcomm
7ca9f46a77 ESSL -> WGSL: Emit func calls, struct access, constants
016ef60d5f ValidateAST asserts when refers to builtIn struct type name
228f56b853 Roll SwiftShader from de870ac7518f to f65e518550a8 (1 revision)
84f88f0032 Roll vulkan-deps from bf2b4e8ba30e to 2e4ed1cfe73c (4 revisions)
d9af242315 Roll Chromium from dd0b2a0bee86 to 9df67ea3cc71 (513 revisions)
dd25e193f2 Add a texture copy test that does multiple layer copies
beb75984ca Fix incorrect verification for SimultaneousUploadAndDraw
e095c06f09 Suppress end2end MultisampledRenderToTexture* failures
6bfd7d139b Manual roll vulkan-deps from c0c0899c3614 to bf2b4e8ba30e (5 revisions)
15b95b4a5d Skip WebGL2UniformBufferTest.LargeArrayOfStructs
5e64ccf9c4 Roll SwiftShader from cea33ab2d5ad to de870ac7518f (1 revision)
45b67e0584 Roll vulkan-deps from 30f8ac1a3f18 to c0c0899c3614 (6 revisions)

Canonical link: https://commits.webkit.org/281691@main
